File: mvp-java/src/main/java/com/wf/mvp/java/impl/Ui.java
package com.wf.mvp.java.impl;

/**
 * MvpMode -> com.wf.mvp.java.imp -> IView
 *
 * @Author: wf-pc
 * @Date: 2020-05-09 20:53
 * <p>
 * Mvp's v-layer specification, used to limit the interfaces that the view must implement.
 */
public interface Ui extends Impl {

    /**
     * Binding P layer instance object.
     * Obtain the generic class type through reflection, and then instantiate the Presenter object.
     */
    void attachPresenter();

    /**
     * Unbind the P layer instance object.
     */
    void detachPresenter();

}

File: mvp-kotlin/src/main/java/com/wf/mvp/kotlin/impl/Ui.kt
package com.wf.mvp.kotlin.impl

/**
 * MvpMode -> com.wf.mvp.kotlin.impl -> IView
 * @Author: wf-pc
 * @Date: 2020-05-10 15:26
 * <p>
 * Mvp's v-layer specification, used to limit the interfaces that the view must implement.
 */
interface Ui : Impl {

    /**
     * Binding P layer instance object.
     * Obtain the generic class type through reflection, and then instantiate the Presenter object.
     */
    fun attachPresenter()

    /**
     * Unbind the P layer instance object.
     */
    fun detachPresenter()

}